sdk/search/azure-search/src/main/java/com/azure/search/DataSources.java

```diff
@@ -0,0 +1,264 @@
+        String description,
+        DataDeletionDetectionPolicy deletionDetectionPolicy) {
+        if (CoreUtils.isNullOrEmpty(name)) {
+            throw new IllegalArgumentException("name cannot be null or empty");
+        }
+        if (CoreUtils.isNullOrEmpty(tableName)) {
+            throw new IllegalArgumentException("tableName cannot be null or empty");
+        }
+        if (CoreUtils.isNullOrEmpty(storageConnectionString)) {
+            throw new IllegalArgumentException("storageConnectionString cannot be null or empty");
+        }
+        return new DataSource()
+            .setName(name)
+            .setType(DataSourceType.AZURE_TABLE)
+            .setCredentials(new DataSourceCredentials()
+                .setConnectionString(storageConnectionString))
+            .setContainer(new DataContainer()
+                .setName(tableName)
+                .setQuery(query))
+            .setDescription(description)
+            .setDataDeletionDetectionPolicy(deletionDetectionPolicy);
+    }
+
+    /**
+     * Creates a new DataSource to connect to an Azure Table.
+     *
+     * @param name The name of the data source.
+     * @param storageConnectionString The connection string for the Azure Storage account.
+     * It must follow this format: "DefaultEndpointsProtocol=https;
+     * AccountName=[your storage account];AccountKey=[your account key];"
+     * Note that HTTPS is required.
+     * @param tableName The name of the Azure table from which to read rows.
+     * @return A new DataSource instance.
+     * @throws IllegalArgumentException if name, tableName or storageConnectionString are null or empty.
+     */
+    public static DataSource azureTableStorage(
+        String name,
+        String storageConnectionString,
+        String tableName) {
+        return DataSources.azureTableStorage(name, storageConnectionString, tableName, null, null, null);
+    }
+
+     /**
+     * Creates a new DataSource to connect to a Cosmos database.
+     *
+     * @param name The name of the data source.
+     * @param cosmosConnectionString The connection string for the Cosmos database. It must follow this format:
+     * AccountName|AccountEndpoint=[your account name or endpoint];
+     * AccountKey=[your account key];Database=[your database name]"
+     * @param collectionName The name of the collection from which to read documents.
+     * @param query Optional. A query that is applied to the collection when reading documents.
+     * @param useChangeDetection Optional. Indicates whether to use change detection when indexing. Default is true.
+     * @param description Optional. Description of the data source
+     * @param deletionDetectionPolicy Optional. The data deletion detection policy for the data source.
+     * @return A new DataSource instance.
+     * @throws IllegalArgumentException if name, collectionName or cosmosConnectionString are null or empty.
+     */
+    public static DataSource cosmos(
```

Comment:
nit: I feel like this and the other related methods should be "cosmosDB" instead of just "cosmos"

Reply:
Based on API board review, it wasdecided on using cosmos and not cusmosDB